File: leetcode_424_Longest Repeating Character Replacement_滑窗算法.py
class Solution(object):
    def characterReplacement(self, s, k):
        res = lo = 0
        counts = collections.Counter()
        for hi in range(len(s)):
            counts[s[hi]] += 1
            max_char_n = counts.most_common(1)[0][1]
            while (hi - lo - max_char_n + 1 > k):
                counts[s[lo]] -= 1
                lo += 1
            res = max(res, hi - lo + 1)
        return res

File: leetcode_424_Longest Repeating Character Replacement_滑窗算法.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int size = s.size();
        vector<int> count(26,0);
        int begin = 0, maxlen = 0, ans = 0;
        for(int end=0;end<size;end++){
        	maxlen = max(maxlen,++count[s[end]-'A']);
        	while(end - begin + 1 - maxlen > k) count[s[begin++] - 'A']--;//相当于count[s[begin] - 'A']--;  begin ++;
        	ans = max(ans,	end - begin + 1);
        }
        return ans;
    }
};